sdk/python/tests/dsl/pipeline_param_tests.py

```diff
@@ -49,4 +49,6 @@ def test_extract_pipelineparam(self):
     self.assertListEqual([p1, p2, p3], params)
     payload = [str(p1) + stuff_chars + str(p2), str(p2) + stuff_chars + str(p3)]
     params = _extract_pipelineparams(payload)
-    self.assertListEqual([p1, p2, p3], params)
\ No newline at end of file
+    self.assertListEqual([p1, p2, p3], params)
+
+  #TODO: add more unit tests to cover real type instances
```

Comment:
nit: as discussed offline, it's better to add a basic e2e test in basic sample folder for the type check feature, which will be verified in the e2e presubmit. It's fine to add it later.

Reply:
Got it. I created an issue to track this: https://github.com/kubeflow/pipelines/issues/974
And, another PR to add some unit tests: https://github.com/kubeflow/pipelines/pull/975